Modules/Applications/AppDomainTransform/app/otbDomainTransform.cxx

```diff
@@ -0,0 +1,453 @@
+
+    AddParameter(ParameterType_InputImage, "in",  "Input Image");
+    SetParameterDescription("in", "This will take an input image to be transformed"
+      " image. For FFT inverse transform, it expects a complex image as two-band"
+      " image in which first band represent real part and second band represent"
+      " imaginary part.");
+
+    AddRAMParameter();
+
+    AddParameter(ParameterType_Choice, "mode", "mode");
+    SetParameterDescription("mode", "This parameter allows one to select between"
+      " fft(fourier) and wavelet");
+    AddChoice("mode.fft", "FFT transform");
+    SetParameterDescription("mode.fft", "FFT transform");
+    AddParameter(ParameterType_Empty, "mode.fft.shift", "false");
+    SetParameterDescription("mode.fft.shift", "Shift transform of fft filter");
+    AddChoice("mode.wavelet", "wavelet");
+    SetParameterDescription("mode.wavelet", "Wavelet transform");
+    AddParameter(ParameterType_Choice, "mode.wavelet.form", "Select wavelet form");
+
+    AddChoice("mode.wavelet.form.haar", "HAAR");
+    AddChoice("mode.wavelet.form.db4", "DAUBECHIES4");
+    AddChoice("mode.wavelet.form.db6", "DAUBECHIES6");
+    AddChoice("mode.wavelet.form.db8", "DAUBECHIES8");
+    AddChoice("mode.wavelet.form.db12", "DAUBECHIES12");
+    AddChoice("mode.wavelet.form.db20", "DAUBECHIES20");
+    AddChoice("mode.wavelet.form.sb24", "SPLINE_BIORTHOGONAL_2_4");
+    AddChoice("mode.wavelet.form.sb44", "SPLINE_BIORTHOGONAL_4_4");
+    AddChoice("mode.wavelet.form.sym8", "SYMLET8");
+
+    //Default value
+    SetParameterString("mode", "wavelet");
+    SetParameterString("mode.wavelet.form", "haar");
+
+    AddParameter(ParameterType_Int,"mode.wavelet.nlevels","Number of decomposition levels");
+    SetParameterDescription("mode.wavelet.nlevels","Number of decomposition levels");
+    SetDefaultParameterInt("mode.wavelet.nlevels",2);
+    SetMinimumParameterIntValue("mode.wavelet.nlevels",2);
+
+    AddParameter(ParameterType_Choice,"dir", "Direction");
+
+    AddChoice("dir.fwd", "Forward");
+    AddChoice("dir.inv", "Inverse");
+
+    AddParameter(ParameterType_OutputImage, "out", "Output Image");
+    SetParameterDescription("out", "This parameter holds the output file name to"
+      " which transformed image will be written. This has a slightly different"
+      " behaviour depending on transform type. \n For Wavelet, output is a"
+      " single band image for both forward and inverse transform. \n For FFT"
+      " forward transform, output is two band image where first band represents"
+      " real part and second band represents imaginary part of a complex image.");
+
+    SetDocExampleParameterValue("in", "input.tif");
+    SetDocExampleParameterValue("mode.wavelet.form", "haar");
+    SetDocExampleParameterValue("out", "output_wavelet_haar.tif");
+    }
+
+  void DoUpdateParameters() ITK_OVERRIDE
+    {
+    // wavelet and fourier are mutually exclusive parameters.
+    // check it here
+#if 0
```

Comment:
this code can be removed, setting the wavelet form and choosing fft mode will not affect the result. The parameters organization is clear enough to understand that the form only applies to wavelet mode